examples/fluids/qfunctions/channel.h

```diff
@@ -64,6 +64,34 @@ CEED_QFUNCTION_HELPER CeedInt Exact_Channel(CeedInt dim, CeedScalar time,
   return 0;
 }
 
+CEED_QFUNCTION_HELPER CeedInt Exact_Channel_Prim(CeedInt dim, CeedScalar time,
+    const CeedScalar X[], CeedInt Nf, CeedScalar q[], void *ctx) {
+
+  const ChannelContext context = (ChannelContext)ctx;
+  const CeedScalar theta0 = context->theta0;
+  const CeedScalar P0     = context->P0;
+  const CeedScalar umax   = context->umax;
+  const CeedScalar center = context->center;
+  const CeedScalar H      = context->H;
+  const CeedScalar cp     = context->newtonian_ctx.cp;
+  const CeedScalar mu     = context->newtonian_ctx.mu;
+  const CeedScalar k      = context->newtonian_ctx.k;
+
+  const CeedScalar y=X[1];
+
+  const CeedScalar Pr    = mu / (cp*k);
+  const CeedScalar Ec    = (umax*umax) / (cp*theta0);
+  const CeedScalar theta = theta0*(1 + (Pr*Ec/3)
+                                   * (1 - Square(Square((y-center)/H))));
+  q[0] = P0;
+  q[1] = umax*(1 - Square((y-center)/H));
+  q[2] = 0.;
+  q[3] = 0.;
+  q[4] = theta;
+
+  return 0;
```

Comment:
Yes, that makes sense. I'll make that change.
Regarding the in/outflow, what do you think of dropping the primitive versions for now if the conservative versions are not tested yet?

Reply:
I think that's fine. I'd like to make all steady inflow (and outflow to the extent relevant) qfunctions just use data anyway, which could come from "imprinting" the initial condition or from sampling a prescribed function. But we should only need one inflow qfunction for conservative and one for primitive. My inclination would be to prescribe state always in terms of primitive variables (since that's a better conditioned way to represent state) and I think the inflow/outflow qfunctions can test the upstream state to determine whether it's sub- or sonic. That would be a future PR.